pkg/clusterfile/clusterfile.go

```diff
@@ -55,17 +63,43 @@ func (c *ClusterFile) GetKubeadmConfig() *kubeadm_config.KubeadmConfig {
 	return &c.kubeadmConfig
 }
 
-func NewClusterFile(path string) (Interface, error) {
-	clusterFileData, err := ioutil.ReadFile(filepath.Clean(path))
+func (c *ClusterFile) SaveAll() error {
+	var configs [][]byte
 
+	cluster, err := yaml.Marshal(c.cluster)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	clusterFile := new(ClusterFile)
-	err = decodeClusterFile(bytes.NewReader(clusterFileData), clusterFile)
+	config, err := yaml.Marshal(c.configs)
+	if err != nil {
+		return err
+	}
+
+	plugin, err := yaml.Marshal(c.plugins)
+	if err != nil {
+		return err
+	}
 
+	kubeconfig, err := yaml.Marshal(c.kubeadmConfig)
 	if err != nil {
+		return err
+	}
+
+	configs = append(configs, cluster, config, plugin, kubeconfig)
```

Comment:
1. rename `configs` to clusterfileBytes.
2. we need the order of cluster file data.
3. return os.NewCommonWriter directly.